pytorch_lightning/plugins/training_type/deepspeed.py

```diff
@@ -37,6 +38,9 @@
 if _DEEPSPEED_AVAILABLE:
     import deepspeed
 
+if TYPE_CHECKING:
+    from pytorch_lightning.plugins.plugins_registry import TrainingTypePluginsRegistry
```

Comment:
We tried this in other PRs and this breaks docs as sphinx does not support forward references :/